iOS-CustomUIButton.flood/Project/Sources/ToggleButton.swift

```diff
@@ -45,11 +47,19 @@ class ToggleButton: UIButton {
 
     @objc
     func toggle() {
+        guard let _ = current else {
+            return
+        }
+        current?.pause()
+        let newTime = current!.duration - current!.time
         if isSelected {
-            reverse?.play()
+            current = reverse
         } else {
-            forward?.play()
+            current = forward
         }
+        current?.reset()
```

Comment:
current?.reset() { 
        current?.offset(to: newTime)
        current?.play()
}